sdk/identity/identity/src/credentials/azurePipelinesCredential.ts

```diff
@@ -118,27 +113,46 @@ export class AzurePipelinesCredential implements TokenCredential {
       logger.error(
         `${credentialName}: Authenticated Failed. Received null token from OIDC request. Response status- ${
           response.status
-        }. Complete response - ${JSON.stringify(response)}`,
+        }. Complete response - ${JSON.stringify(response)}`
       );
-      throw new CredentialUnavailableError(
+      throw new AuthenticationError(
+        response.status,
         `${credentialName}: Authenticated Failed. Received null token from OIDC request. Response status- ${
           response.status
-        }. Complete response - ${JSON.stringify(response)}`,
+        }. Complete response - ${JSON.stringify(response)}`
       );
     }
-    const result = JSON.parse(text);
-    if (result?.oidcToken) {
-      return result.oidcToken;
-    } else {
-      logger.error(
-        `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
-          result,
-        )}`,
-      );
-      throw new CredentialUnavailableError(
-        `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
-          result,
-        )}`,
+    try {
+      const result = JSON.parse(text);
+      if (result?.oidcToken) {
+        return result.oidcToken;
+      } else {
+        if (response.status !== 200) {
+          logger.error(
+            `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
+              result
+            )}`
+          );
+          throw new AuthenticationError(
+            response.status,
+            `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
+              result
+            )}`
+          );
+        } else {
+          logger.error(
```

Comment:
Nit, feel free to take it or leave it

I agree with (IIRC) @billwert 's point about this being a service bug not something that the SDK should be concerned with. In general I do not code against specific service bugs, more of a single "I expected an OIDC token and did not get one. here's what I got" validation will be sufficient.

Either way, the status code is an implementation detail and not very useful for users of the SDK IMO so we can combine these two errors into one and not worry about the status code (although we could log it).

Reply:
which errors should i combine?

Reply:
On line 133 we check if the response status is _not_ 200 - and we throw an error 
On line 145 we have an else (so the response status is 200) - and we throw an error

I think we remove the if/else entirely and just throw a single error if we're unable to get the oidcToken. The error message or the logger.error messages could include the response.status if we need but we don't need to maintain two branches - does that clarify?

Reply:
Yes, I tried to simplify that part now. Can you check?

Reply:
this is verified offline